src/infra/src/table/dashboards.rs

```diff
@@ -254,34 +246,25 @@ pub async fn delete_all() -> Result<(), errors::Error> {
 }
 
 /// Tries to get a dashboard ORM entity and its parent folder ORM entity.
-async fn get_model(
-    db: &DatabaseConnection,
+async fn get_model<C: ConnectionTrait>(
+    db: &C,
     org_id: &str,
-    folder_id: &str,
     dashboard_id: &str,
-) -> Result<Option<(folders::Model, Option<dashboards::Model>)>, sea_orm::DbErr> {
-    let select_folders = folders::Entity::find()
-        .filter(folders::Column::Org.eq(org_id))
-        .filter(folders::Column::Type.eq::<i16>(FolderType::Dashboards.into()))
-        .filter(folders::Column::FolderId.eq(folder_id));
-
-    let Some(folder) = select_folders.one(db).await? else {
-        return Ok(None);
-    };
-
-    let maybe_dashboard = folder
-        .find_related(dashboards::Entity)
+) -> Result<Option<(folders::Model, dashboards::Model)>, sea_orm::DbErr> {
+    let f_and_d = dashboards::Entity::find()
         .filter(dashboards::Column::DashboardId.eq(dashboard_id))
+        .find_also_related(folders::Entity)
+        .filter(folders::Column::Org.eq(org_id))
         .one(db)
-        .await?;
-
-    Ok(Some((folder, maybe_dashboard)))
+        .await?
+        .and_then(|(d, maybe_f)| maybe_f.map(|f| (f, d)));
+    Ok(f_and_d)
 }
 
 /// Lists dashboard ORM models using the given parameters. Returns each
 /// dashboard and its parent folder.
-async fn list_models(
-    db: &DatabaseConnection,
+async fn list_models<C: ConnectionTrait>(
+    db: &C,
```

Comment:
Swap the concrete `DatabaseConection` type for the `ConnectionType` trait so we can pass in a transaction.

This isn't strictly necessary for this PR, but seems like a good pattern to follow in general for anywhere where we need to pass a DB connection as a param